modules/workload-identity/main.tf

```diff
@@ -0,0 +1,32 @@
+resource "azurerm_user_assigned_identity" "workload_identity" {
+  name                = var.identity_name
+  resource_group_name = var.resource_group_name
+  location            = var.location
+}
```

Comment:
Why is workload identity required? Is there a way to do it without it?

Reply:
If you want to do it without workload identity, the node role would need permissions to the registry and object store. 

This is a least access privilege approach to accomplish this and an assumption already baked into the launch arch. 

++ @bcsherma if you want to add more context.